tests/components/igloohome/test_config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Test the igloohome config flow."""
+
+from unittest.mock import AsyncMock, patch
+
+from aiohttp import ClientError
+from igloohome_api import AuthException
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.igloohome.const import DOMAIN
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+
+async def test_form(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, auth_successful
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with auth_successful:
```

Comment:
```suggestion
```
should be unnecessary

Reply:
Hey @zweckj . By removing this, fixture does not patch the Auth class and the test ends up trying to execute a HTTP request that fails. Not quite sure why that happens though... Any ideas? 

Attached the failing test output.

```
============================== 1 failed in 0.42s ===============================
FAILED                                    [100%]INFO:homeassistant.loader:Loaded igloohome from homeassistant.components.igloohome

tests/components/igloohome/test_config_flow.py:15 (test_form)
hass = <HomeAssistant RUNNING>
mock_setup_entry = <AsyncMock name='async_setup_entry' id='281472760325440'>
auth_successful = <unittest.mock._patch object at 0xffff7b909850>

    async def test_form(
        hass: HomeAssistant, mock_setup_entry: AsyncMock, auth_successful
    ) -> None:
        """Test we get the form."""
        result = await hass.config_entries.flow.async_init(
            DOMAIN, context={"source": config_entries.SOURCE_USER}
        )
        assert result["type"] == FlowResultType.FORM
        assert result["errors"] == {}
    
>       result = await hass.config_entries.flow.async_configure(
            result["flow_id"],
            {
                CONF_CLIENT_ID: "test-client-id",
                CONF_CLIENT_SECRET: "test-client-secret",
            },
        )

test_config_flow.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../homeassistant/data_entry_flow.py:341: in async_configure
    result = await self._async_configure(flow_id, user_input)
../../../homeassistant/data_entry_flow.py:388: in _async_configure
    result = await self._async_handle_step(
../conftest.py:761: in _flow_manager_async_handle_step
    result = await _original_flow_manager_async_handle_step(self, flow, *args)
../../../homeassistant/data_entry_flow.py:491: in _async_handle_step
    result: _FlowResultT = await getattr(flow, method)(user_input)
../../../homeassistant/components/igloohome/config_flow.py:44: in async_step_user
    await auth.async_get_access_token()
../../../venv/lib/python3.12/site-packages/igloohome_api/__init__.py:63: in async_get_access_token
    response = await self.session.post(
../../../venv/lib/python3.12/site-packages/aiohttp/client.py:701: in _request
    conn = await self._connector.connect(
../../../venv/lib/python3.12/site-packages/aiohttp/connector.py:544: in connect
    proto = await self._create_connection(req, traces, timeout)
../../../venv/lib/python3.12/site-packages/aiohttp/connector.py:1050: in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
../../../venv/lib/python3.12/site-packages/aiohttp/connector.py:1363: in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
../../../venv/lib/python3.12/site-packages/aiohttp/connector.py:1109: in _wrap_create_connection
    sock = await aiohappyeyeballs.start_connection(
../../../venv/lib/python3.12/site-packages/aiohappyeyeballs/impl.py:109: in start_connection
    raise first_exception
../../../venv/lib/python3.12/site-packages/aiohappyeyeballs/_staggered.py:127: in run_one_coro
    result = await coro_fn()
../../../venv/lib/python3.12/site-packages/aiohappyeyeballs/impl.py:151: in _connect_sock
    sock = socket.socket(family=family, type=type_, proto=proto)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'pytest_socket.disable_socket.<locals>.GuardedSocket'>
family = <AddressFamily.AF_INET: 2>, type = <SocketKind.SOCK_STREAM: 1>
proto = 6, fileno = None

    def __new__(cls, family=-1, type=-1, proto=-1, fileno=None):
        if _is_unix_socket(family) and allow_unix_socket:
            return super().__new__(cls, family, type, proto, fileno)
    
>       raise SocketBlockedError()
E       pytest_socket.SocketBlockedError: A test tried to use socket.socket.

../../../venv/lib/python3.12/site-packages/pytest_socket.py:89: SocketBlockedError
WARNING:tests.conftest:Lingering timer after test <TimerHandle when=137813.398080583 _set_result(<Future finished result=None created at /usr/local/lib/python3.12/asyncio/base_events.py:449>) at /IdeaProjects/homeassistant-core/venv/lib/python3.12/site-packages/aiohappyeyeballs/_staggered.py:20 created at /IdeaProjects/homeassistant-core/venv/lib/python3.12/site-packages/aiohappyeyeballs/_staggered.py:152>
WARNING:tests.conftest:Lingering timer after test <TimerHandle when=137813.398995083 _set_result(<Future finished result=None created at /usr/local/lib/python3.12/asyncio/base_events.py:449>) at /IdeaProjects/homeassistant-core/venv/lib/python3.12/site-packages/aiohappyeyeballs/_staggered.py:20 created at /IdeaProjects/homeassistant-core/venv/lib/python3.12/site-packages/aiohappyeyeballs/_staggered.py:152>
WARNING:tests.conftest:Lingering timer after test <TimerHandle when=137813.399968458 _set_result(<Future finished result=None created at /usr/local/lib/python3.12/asyncio/base_events.py:449>) at /IdeaProjects/homeassistant-core/venv/lib/python3.12/site-packages/aiohappyeyeballs/_staggered.py:20 created at /IdeaProjects/homeassistant-core/venv/lib/python3.12/site-packages/aiohappyeyeballs/_staggered.py:152>
WARNING:tests.conftest:Lingering timer after test <TimerHandle when=137813.401272958 _set_result(<Future finished result=None created at /usr/local/lib/python3.12/asyncio/base_events.py:449>) at /IdeaProjects/homeassistant-core/venv/lib/python3.12/site-packages/aiohappyeyeballs/_staggered.py:20 created at /IdeaProjects/homeassistant-core/venv/lib/python3.12/site-packages/aiohappyeyeballs/_staggered.py:152>
WARNING:tests.conftest:Lingering timer after test <TimerHandle when=137813.402235791 _set_result(<Future finished result=None created at /usr/local/lib/python3.12/asyncio/base_events.py:449>) at /IdeaProjects/homeassistant-core/venv/lib/python3.12/site-packages/aiohappyeyeballs/_staggered.py:20 created at /IdeaProjects/homeassistant-core/venv/lib/python3.12/site-packages/aiohappyeyeballs/_staggered.py:152>

Process finished with exit code 1
```